### PR TITLE
Remove cross-signing signature validation

### LINKED ISSUE #12548: Is it correct that Synapse validates cross-signing signatures?
When you upload a cross-signing signature, Synapse attempts to validate that signature. 

https://spec.matrix.org/v1.2/client-server-api/#cross-signing doesn't seem to require this (though it does document `M_INVALID_SIGNATURE: For example, the self-signing or user-signing key had an incorrect signature`), and it's not obvious it's the homeserver's job to do this validation.

### PR DESCRIPTION
**Pull request by marekvospel**
This Pull request removes signature validation as discusses in #12548.

closes #12548 

### Pull Request Checklist

* [X] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [X] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [X] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

Signed-off-by: Marek Vospěl <marek@vospel.cz>

### CODE REVIEW

**Comment by marekvospel** [comment]
I only removed it from upload_signatures_for_device_keys, as it seems like the problem in #12548. It might still be useful to keep it in upload_keys_for_user.

It might still be useful to have some sort of validation, like checking the keys length etc. just like dendrite has (sanityCheckKey), so users don't upload totally random data as signatures.
https://github.com/matrix-org/dendrite/blob/11b557097c6745309c09b58f681080d3fcc4f9f5/userapi/internal/cross_signing.go#L34

All of those are just thoughts, it won't be hard to remove the validation completely.

**Comment by marekvospel** [comment]
mb, wanted to merge, not rebase... No changes have been made

**Comment by marekvospel** [comment]
Any idea why the test is failing? When running locally with the same options(focal, multi postgres, workers, asyncio), I get no errors... 
`docker run --rm -it -v ./:/src -v ./logs:/logs -e ASYNCIO_REACTOR=true -e MULTI_POSTGRES=true -e WORKERS=true matrixdotorg/sytest-synapse:focal`

Most of the tests seem to fail because of `Timed out waiting for test at /sytest/run-tests.pl line 864.`

**Comment by clokep** [comment]
It probably flaked. I ran it again.

File: synapse/handlers/e2e_keys.py [hunk 1]
```diff
@@ -1107,10 +1107,6 @@ async def _process_self_signatures(
                     # can skip it, since it should be exactly the same
                     continue
 
-                _check_device_signature(
-                    user_id, self_signing_verify_key, device, stored_device
-                )
```
Comment (clokep) [inline synapse/handlers/e2e_keys.py:1112]: There's one more caller of this in `_check_master_key_signature` -- is that use-case valid?

Comment (marekvospel) [reply synapse/handlers/e2e_keys.py:1112]: As I mentioned in a comment, I left the validation in /keys/upload and I do belive that call is necessary.
(as in cross signing master key is signed by the device keys)
master key -> self signing key -> device key -> master key

Comment (clokep) [reply synapse/handlers/e2e_keys.py:1112]: Ah-ha, I didn't connect that `/keys/upload` called into that directly. 👍

**Comment by marekvospel** [comment]
Just thinking about it now, I don't think any of these changes will fix anything in the communication between synapse and dendrite, as there was no validation in /keys/query nor in the federation handler. Keeping the validation might still be an option.

I'll do some debugging on /keys/query endpoint and post an update on whether my hyphotheis is correct.

**Comment by marekvospel** [comment]
I've done some testing and even with validation for `/keys/upload`, I'm still running into #12548.

That does make sense, as the validation was only for `/keys/upload` and `/keys/signatures/upload`, which from my understanding are only used by users of the current homeserver, so it is unrelated to federation.

**Pull request closed without merging**